gradle-baseline-java-config/resources/checkstyle/checkstyle.xml

```diff
@@ -334,6 +334,18 @@
             <property name="message" value="printStackTrace is not generally allowed."/>
             <property name="ignoreComments" value="true"/>
         </module>
+        <module name="RegexpSinglelineJava">
+            <property name="id" value="BanSystemOut"/>
+            <property name="format" value="System\.out\."/>
+            <property name="message" value="System.out usage is not generally allowed. Please use appropriate logging methods"/>
```

Comment:
I still don't think this error message gives an uninformed user enough information about what they should do (or why it's important). What about something like:

```
Logging with System.out is not allowed because it has no metadata and can't be configured at runtime. Please use an SLF4J logger instead, e.g. log.info("Message").
```